File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
def is_group_word(word):
    prev = word[0]
    before = set()
    isGroup = True
    
    for i in range(len(word)):
        if prev != word[i]:
            before.add(prev)
        
        if word[i] in before:
            isGroup = False
        
        prev = word[i]
    
    return isGroup

N = int(input())

words = [input().strip() for _ in range(N)]

ans = 0

for word in words:
    if is_group_word(word):
        ans += 1

print(ans)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int N = sc.nextInt();
        int count = 0;
        
        for (int i = 0; i < N; i++) {
            String word = sc.next();
            if (isGroupWord(word))
                count++;
        }
        sc.close();
        System.out.print(count);
    }
    
    private static boolean isGroupWord(String word) {
        char prev = word.charAt(0);
        Set<Character> before = new HashSet<>();
        boolean isGroup = true;
        
        for (int i = 0; i < word.length(); i++) {
            char cur = word.charAt(i);
            if (prev != cur)
                before.add(prev);
            if (before.contains(cur)) {
                isGroup = false;
                break;
            }
            prev = cur;
        }
        return isGroup;
    }
}